app/frontend/src/features/connections/friends/AddFriendButton.tsx

```diff
@@ -0,0 +1,57 @@
+import { makeStyles } from "@material-ui/core";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import React from "react";
+import { useQueryClient, useMutation } from "react-query";
+import { SetMutationError } from ".";
+import Button from "../../../components/Button";
+import { PersonAddIcon } from "../../../components/Icons";
+import { service } from "../../../service";
+
+const useStyles = makeStyles((theme) => ({
+  editButton: {
+    marginBottom: theme.spacing(2),
+  },
+}));
+
+interface AddFriendButtonProps {
+  setMutationError: SetMutationError;
+  userId: number;
+}
+
+export default function AddFriendButton({
+  setMutationError,
+  userId,
+}: AddFriendButtonProps) {
+  const classes = useStyles();
+  const queryClient = useQueryClient();
+
+  const { isLoading, mutate: sendFriendRequest, reset } = useMutation<
```

Comment:
Not sure if this hook will be re-usable anywhere else, and it seems simple enough to me that I think it might be ok to just inline it in this component, which is also not meant to be super re-usable since it's a specialised button for triggering the "add friend" action